### DISCUSSION TITLE
Music player should shuffle music in the same order that beatmaps are displayed on the song select

### DISCUSSION THREAD

**Discussion by OwenCMYK** [ppy/osu#27345] [Ideas]
I find it really annoying when I go to the title screen and during that time a new song starts playing. Then when I go back to song select I press left (hoping to get back to the song I was previously on), but the music player and song select use a completely different order if you have your sorting changed (which I do), so naturally this ends up selecting a completely random song and I have to search for the song I was previously on. This sounds incredibly minor but comes up weirdly often. So I propose that the music shuffle should order songs the same as whichever sorting mode the song select is currently using.

I also think when selecting a beatmap collection in the dropdown, the song player should be limited to songs from that collection, allowing you to use them as playlists.

**Comment by Gabixel** [comment]
I honestly don't know if I see any sense in it: the music player and the carousel are technically two separate entities, and syncing the beatmap sorting on both (by taking it from the carousel) seems "intrusive"/out of place. There's also a (no longer) proposal to [add a generic shuffle option for the music player](https://github.com/ppy/osu/discussions/13039), which could potentially screw up this idea even more.

I will admit, however, that this scenario used to happen to me as well, but I don't think it's that nerve-wracking, and using the "date added" sorting (and/or using the music player itself on the carousel to go backwards/forward) is probably the cleanest solution (for me).

**Reply by OwenCMYK** [reply]
You've also just brought to my attention that when selecting a beatmap collection for the music player doesn't limit the player to playing songs of that collection (which I feel that it should so that you can use collections as playlists). They could simply make it so the shuffle option just ignores sorting order and plays in a random order. People who want shuffle probably don't have this as their priority anyway.
